src/open_producten/producttypen/tests/api/test_producttype.py

```diff
@@ -957,3 +1100,61 @@ def test_en_content_and_fallback(self):
                 },
             ],
         )
+
+
+class TestProductTypeFilterSet(BaseApiTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.path = reverse("producttype-list")
+
+    def test_gepubliceerd_filter(self):
+        ProductTypeFactory.create(gepubliceerd=True)
+        ProductTypeFactory.create(gepubliceerd=False)
+
+        response = self.client.get(self.path + "?gepubliceerd=true")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 1)
+
+    def test_uniforme_product_naam_filter(self):
+        ProductTypeFactory.create(
+            uniforme_product_naam=UniformeProductNaamFactory(naam="parkeervergunning")
+        )
+        ProductTypeFactory.create(
+            uniforme_product_naam=UniformeProductNaamFactory(naam="aanleunwoning")
+        )
+
+        response = self.client.get(
+            self.path + "?uniforme_product_naam=parkeervergunning"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 1)
+
+    def test_externe_code_filter(self):
+        product_type_1 = ProductTypeFactory.create()
+        ExterneCodeFactory(naam="ISO", code="12345", product_type=product_type_1)
+
+        product_type_2 = ProductTypeFactory.create()
+        ExterneCodeFactory(naam="ISO", code="9837549857", product_type=product_type_2)
+
+        response = self.client.get(self.path + "?externe_code=[ISO:12345]")
```

Comment:
You can pass the query parameters as a dictionary, this applies for the other tests as well
```suggestion
        response = self.client.get(self.path, {"externe_code": "[ISO:12345]"})
```